Support for non-persistent sessions and table filtering

hegupta-adbe opened the pull request:
Please ensure your pull request adheres to the following guidelines:
- [ ] make sure to link the related issues in this description
- [ ] when merging / squashing, make sure the fixed issue references are visible in the commits, for easy compilation of release notes

## Related Issues


Thanks for contributing!

dominique-pfister commented on src/OneDrive.js:
```diff
@@ -423,7 +423,11 @@ export class OneDrive {
+    return this.getWorkbookFromPath(`/drives/${driveItem.parentReference.driveId}/items/${driveItem.id}/workbook`);
+  }
+
+  getWorkbookFromPath(path) {
```
what is the use-case of that method?

hegupta-adbe replied:
Githib actions implementation for post-submit processing. Franklin passes us the raw path in the event payload and we'd like to create a workbook object out of that https://github.com/hegupta-adbe/form-data-processor/blob/helix-onedrive/index.js#L202

dominique-pfister replied:
I'd rather use the [OneDrive.getDriveItem](https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L150) API followed by [OneDrive.getWorkbook](https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L190). Your code change would return a `Workbook` instance with an opaque path that has neither been validated nor tested for existence.

tripodsan replied:
or at least: https://github.com/adobe/helix-onedrive-support/blob/eff748fa9c1daeacd9c97594241a0e607da0ecae/src/OneDrive.d.ts#L112C10-L112C26

dominique-pfister commented on src/excel/Workbook.js:
```diff
@@ -23,13 +23,15 @@ export class Workbook extends NamedItemContainer {
     const uri = `${this.uri}/createSession`;
     const result = await this._oneDrive.doFetch(uri, false, {
       method: 'POST',
+      headers: {'Content-Type': 'application/json'},
+      body: JSON.stringify({persistChanges: persistent})
```
wrong linting, space required before `}` and after `{`

dominique-pfister commented on src/excel/Workbook.js:
```diff
@@ -23,13 +23,15 @@ export class Workbook extends NamedItemContainer {
   }
 
-  async createSession() {
+  async createSession(persistent = true) {
```
if you call the parameter `persistChanges`, you can simply write:
```
body: JSON.stringify({ persistChanges })
```
below

hegupta-adbe commented:
@tripodsan @dominique-pfister I've incorporated the comments and fixed all linting errors, and all checks are passing now. Please review and merge if it looks ok now.
